### PR TITLE
eww: add terminal integration options

### PR DESCRIPTION
**Pull request by Jeansidharta**
### Description

Use eww's shell-completions command to generate completions for bash, zsh and fish.

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all` or `nix develop --ignore-environment .#all` using Flakes.

- [ ] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://nix-community.github.io/home-manager/#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

#### Maintainer CC

@mainrs

### CODE REVIEW

File: modules/programs/eww.nix [hunk 1]
```diff
@@ -30,10 +31,34 @@ in {
         {file}`$XDG_CONFIG_HOME/eww`.
       '';
     };
+
+    enableBashIntegration = mkEnableOption "Bash integration";
```
Comment (mainrs) [inline modules/programs/eww.nix:35]: Can you enable these completion options by default, please?

File: modules/programs/eww.nix [hunk 2]
```diff
@@ -30,10 +31,34 @@ in {
...
+
+    enableZshIntegration = mkEnableOption "Zsh integration";
```
Comment (mainrs) [inline modules/programs/eww.nix:37]: Same.

File: modules/programs/eww.nix [hunk 3]
```diff
@@ -30,10 +31,34 @@ in {
...
+
+    enableFishIntegration = mkEnableOption "Fish integration";
```
Comment (mainrs) [inline modules/programs/eww.nix:39]: Same.

**Review by mainrs — CHANGES_REQUESTED** [review]
It would be good to see if eww also has completions for other shells like nushell. If we're already at it.

**Comment by Jeansidharta** [comment]
When running `eww shell-completions --shell` on version 0.6.0 I got this message:

```
[❯] eww shell-completions --shell
error: a value is required for '--shell <SHELL>' but none was supplied
  [possible values: bash, elvish, fish, powershell, zsh]

For more information, try '--help'.
```

- elvish is not in the home-manager modules
- powershell is windows only.

So, no nushell or anything else to implement, unfortunately.

**Comment by Jeansidharta** [comment]
Just enabled them by default. I didn't do this at first with a fear of breaking some users environment. But this should be fine? Probably?

**Review by mainrs — APPROVED** [review]
> Just enabled them by default. I didn't do this at first with a fear of breaking some users environment. But this should be fine? Probably?

Good question. I just took a look at other modules and most of them enabled them by default. I guess theoretically this will override existing completions.

LGTM. I think this is ready to be merged.

**Comment by rycee** [comment]
Thanks! Merged to master now 🙂

**Pull request merged**